### DISCUSSION TITLE
Help wanted: error when using STL libs

### DISCUSSION THREAD

**Discussion by dudozermaks** [clangd/clangd#1457] [Q&A] [answered]
Hi everyone!
I have project created with CubeIDE for stm32f411. Project is in c++, every c++ file named correctly (with cpp or hpp extension) and compiles with g++.
To edit code in external editor i exported compile_commands.json and placed it into root folder of my project. Everything works fine: auto-completion, no errors. But, when include vector library, error occurs: 
![Selection_052](https://user-images.githubusercontent.com/64933915/211214682-df2434d2-7018-4862-ba75-b2380f427ea1.png)
I tried to include other standard libraries, same result:
![Selection_053](https://user-images.githubusercontent.com/64933915/211214704-5ffbd825-74c8-4e60-9a15-96ed57836b25.png)
**After including vector and savingf file, i compiled the project and it compiles fine, with no errors.**
As you can see, i am including vector lib in header file, maybe if i do that in .cpp file, it will be fine? no, same error.
Maybe compiled_commands.json has `-nostdlib` flag?
![Selection_054](https://user-images.githubusercontent.com/64933915/211214891-858437d5-72f9-4c24-80ed-98119ea4d073.png)
Nope, it has not.
I use Neovim + [Mason](https://github.com/williamboman/mason.nvim) (for installing LSPs) + clangd. I have no other LSPs installed.
Attaching [compile_commands.txt](https://github.com/clangd/clangd/files/10369312/compile_commands.txt) (renamed file because GitHub does not support JSON files)
Any ideas? Thanks in advance!

**Comment by HighCommander4** [comment]
Have a look at https://clangd.llvm.org/troubleshooting#cant-find-standard-library-headers-map-stdioh-etc. Using `--query-driver` is usually needed for cross-compiler scenarios like this one.

**Reply by dudozermaks** [reply]
I tried that, added --query-driver=/usr/bin/arm-none-eabi* to flags for clangd. it doesn't help. I found out that, when i delete compile_commands.json, there are no errors when i include STL libs, so the problem is in compile_commands.json, i believe.

**Comment by dudozermaks** [comment, marked as answer]
I found out the problem, i think. I had following lines in compile_commands.json:
```json
[
  {
    "arguments": [
      "/usr/bin/arm-none-eabi-gcc",
      ... My arguments
    ],
    "directory": "ProjectDirectory/Debug",
    "file": "ProjectDirectory/Drivers/STM32F4xx_HAL_Driver/Src/stm32f4xx_hal_dma.c",
    "output": "ProjectDirectory/Debug/Drivers/STM32F4xx_HAL_Driver/Src/stm32f4xx_hal_dma.o"
  },
```
I deleted every argument except `"/usr/bin/arm-none-eabi-gcc"`, started neovim, aaand error was still showing. I put everything back, as it was, but changed `arm-none-eabi-gcc` to `gcc` and `arm-none-eabi-g++` to `g++` and it works fine now! I have both my headers and STL!

**Reply by HighCommander4** [reply]
Cool, it sounds like you got it to work by getting clangd to use the default standard library installed on your system (as opposed to the one that ships with your cross compiler).
